### PR TITLE
 vimPlugins.gruvbox-nvim: init at 2021-04-23

### PR DESCRIPTION
**Pull request by jupblb**
###### Motivation for this change
Added https://github.com/npxbr/gruvbox.nvim along with an override for it's dependency, https://github.com/rktjmp/lush.nvim (see https://github.com/npxbr/gruvbox.nvim#installing)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/generated.nix [hunk 1]
```diff
@@ -5332,12 +5332,12 @@ let
...
     pname = "vim-clap";
-    version = "2021-04-17";
+    version = "2021-04-25";
     src = fetchFromGitHub {
       owner = "liuchengxu";
       repo = "vim-clap";
-      rev = "8e13b23d69549c95d9c223ea5c2487d5dd9558f7";
-      sha256 = "1biiq07dhrz9vhk0yg3zkkv3329nyla6lp8kavdzqrvqg0hsbr2j";
+      rev = "1afdd263a862bae0641f565e3f2952e1c01cec43";
+      sha256 = "0c2jrz02dsdykc3xxqw1yfnllmrpwzs6ygjqcclghw5mygfc3xcg";
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/generated.nix:5352]: looks like you need to update the `cargoSha256` in overrrides

```
builder for '/nix/store/jnav35qdfblsk868q0m3q1gdgvv1sg78-maple.drv' failed with exit code 1; last 10 log lines:

  ERROR: cargoSha256 is out of date

  Cargo.lock is not the same in maple-vendor.tar.gz

  To fix the issue:
  1. Use "0000000000000000000000000000000000000000000000000000" as the cargoSha256 value
  2. Build the derivation and wait for it to fail with a hash mismatch
  3. Copy the 'got: sha256:' value back into the cargoSha256 field
cannot build derivation '/nix/store/aybp3inz3gg64955pk3fb2jyv0qacyg6-vimplugin-vim-clap-2021-04-25.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/k2zykjrd15fb8pgxzp91jarzj0yrscki-env.drv': 1 dependencies couldn't be built
error: --- Error ------------------------------------------------------------------------------------------------------- nix
build of '/nix/store/k2zykjrd15fb8pgxzp91jarzj0yrscki-env.drv' failed
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120629` at 1f7f1954 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e6fb517e4e5ab542674df74699ee1b8e">vimPlugins.vim-clap</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.vim-hexokinase</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/misc/vim-plugins/generated.nix (hunk 1 above)
Comment (jupblb) [reply pkgs/misc/vim-plugins/generated.nix:5352]: Ouch :/ Fixed

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -273,6 +273,10 @@ self: super: {
     dependencies = with self; [ plenary-nvim ];
   });
 
+  gruvbox-nvim = super.gruvbox-nvim.overrideAttrs (old: {
+    dependencies = with super; [ lush-nvim ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:277]:
```suggestion
    dependencies = with self; [ lush-nvim ];
```

Comment (jupblb) [reply pkgs/misc/vim-plugins/overrides.nix]: Right, I missed that (I'm a bit fuzzy on `super`/`self`). Thanks for spotting this. Fixed. :)

**Comment by jupblb** [comment]
Resolved merge conflict after https://github.com/NixOS/nixpkgs/pull/121092

**Pull request merged**